src/android/AES.java

```diff
@@ -16,16 +16,16 @@
     private static final String CIPHER_MODE = "GCM";
     private static final int KEY_SIZE = 256;
     private static final int VERSION = 1;
-    private static final Cipher CIPHER = getCipher();
+    private static final Cipher GLOBAL_CIPHER = getCipher(CIPHER_MODE);
 
     public static JSONObject encrypt(byte[] msg, byte[] adata) throws Exception {
         byte[] iv, ct, secretKeySpec_enc;
-        synchronized (CIPHER) {
+        synchronized (GLOBAL_CIPHER) {
             SecretKeySpec secretKeySpec = generateKeySpec();
             secretKeySpec_enc = secretKeySpec.getEncoded();
-            initCipher(Cipher.ENCRYPT_MODE, secretKeySpec, null, adata);
-            iv = CIPHER.getIV();
-            ct = CIPHER.doFinal(msg);
+            initCipher(Cipher.ENCRYPT_MODE, secretKeySpec, null, adata, GLOBAL_CIPHER);
+            iv = GLOBAL_CIPHER.getIV();
+            ct = GLOBAL_CIPHER.doFinal(msg);
         }
 
         JSONObject value = new JSONObject();
```

Comment:
@mibrito707  - no need for a README statement, how to use the mode, IMHO.

As you can see in line 35 (`value.put("mode", CIPHER_MODE);`) the cipher mode is already stored in the corresponding JSON object.  
For legacy keys obviously the field is not `GCM` but `CCM`.  
Therefore the `data.getString("mode")));` call in `src/android/SecureStorage.java:161` takes the stored mode - either new GCM or legacy CCM values.